utils/azureResourceType.d.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+/**
+ * Normalized type for Azure resources that uniquely identifies resource type for the purposes
+ * of the Azure extensions
+ */
+export declare enum AzureResourceType {

```

Comment:
Although this is Azure-specific stuff, and so _technically_ probably doesn't belong in this package (but in `azureutils` instead), it does not add any dependencies and the host API needs to reference this type. I'd rather not move the host API to the `azureutils` package, so that leaves here as the choice.

Reply:
Personally I think of having e.g. `AzureResourceType.StaticWebApp` as a feature, as it feels more `const`-y, and using it as a value could have good applications. But since TypeScript does compile-time checks of both `type` and `enum` values, I'm not sure there's any compile-time advantage there.